### PR TITLE
Remove Safe

### PR DESCRIPTION
**Pull request by rodnaph**
**Summary**

Removes `thecodingmachine/safe` from the dependencies [as suggested](https://github.com/thecodingmachine/gotenberg-php-client/pull/21#issuecomment-751654152) for PHP 7.1 support.

**Test plan (required)**

The existing test suite should remain green.

**Checklist**

- [X] Have you followed the guidelines in our [CONTRIBUTING](CONTRIBUTING.md) guide?
- [X] Have you lint your code locally prior to submission (`make lint`)?
- [X] Have you written new tests for your core changes, as applicable?
- [X] Have you successfully ran tests with your changes locally (`make tests`)?
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code

### CODE REVIEW

**Comment by rodnaph** [comment]
@gulien What's the deal with the codecov/travis-ci build steps, should they be running? Looks like the travis one passed on the last PR merged, but something has happened with a migration to travis-ci.com (I'm not familiar with this tool sorry).

https://travis-ci.com/github/thecodingmachine/gotenberg-php-client

**Comment by gulien** [comment]
That's because Travis uses a "credit" system, even for OSS, and thecodingmachine organization does not have credits anymore.

I guess I'll need to migrate to GitHub actions 😄

**Comment by rodnaph** [comment]
> I guess I'll need to migrate to GitHub actions 😄

I'll raise another PR... 🤣

**Comment by rodnaph** [comment]
@gulien I've moved the build to Github Actions, but it's not running on this PR as I believe you need to specifically enable it to run on forks in the repo settings. You can see the builds run on my repo though:

https://github.com/rodnaph/gotenberg-php-client/actions/runs/449204547

There are a bunch of failed builds in there too so you can see the build fails when steps fail. A tweak was needed to resolve alternative DNS for the Gotenberg service in the `ClientTest`.

**Comment by rodnaph** [comment]
@gulien Your suggested Github Actions changes are on this branch now.

eg. https://github.com/rodnaph/gotenberg-php-client/runs/1621077191?check_suite_focus=true

File: src/Client.php [hunk 1]
```diff
@@ -57,6 +57,7 @@ public function store(GotenbergRequestInterface $request, string $destination):
         $response = $this->handleResponse($this->client->sendRequest($this->makeMultipartFormDataRequest($request)));
         $fileStream = $response->getBody();
         $fp = fopen($destination, 'w');
+        assert(is_resource($fp));
         fwrite($fp, $fileStream->getContents());
```
Comment (rodnaph) [inline src/Client.php]: @gulien Without `Safe` the calls to `fwrite`, `fclose` etc... now don't check the result isn't `false`, do you think it's worth adding these checks here instead?

Comment (gulien) [reply src/Client.php]: I think it is worth adding those checks indeed ; fail fast and fail loud 👍

Comment (rodnaph) [reply src/Client.php]: Done, I borrowed the `FilesystemException` class from Safe.

**Review by gulien — APPROVED** [review]
:+1:

**Comment by gulien** [comment]
Thank your very much @rodnaph 😉

**Pull request merged**